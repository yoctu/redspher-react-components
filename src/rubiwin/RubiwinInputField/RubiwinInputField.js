import React from 'react'
import Input from '@material-ui/core/Input'
import PropTypes from 'prop-types'
import style from './RubiwinInputField.module.scss'

import { StylesProvider } from '@material-ui/core/styles'

/** this is a simple input
 * full doc: https://material-ui.com/api/input/
 * */

const RubiwinInputField = ({
  className = '',
  id,
  defaultValue,
  disabled = false,
  endAdornment,
  error = false,
  fullWidth = false,
  inputProps,
  inputRef,
  multiline = false,
  name,
  onChange,
  placeholder,
  readOnly,
  required,
  rows,
  rowsMax,
  startAdornment,
  type,
  value
}) => (
  <StylesProvider injectFirst>
    <Input
      className={`${className}`}
      classes={{
        root: style.root,
        focused: style.focused,
        disabled: style.disabled,
        error: style.error
      }}
      disableUnderline
      endAdornment={endAdornment}
      id={id}
      defaultValue={defaultValue}
      disabled={disabled}
      error={error}
      fullWidth={fullWidth}
      inputProps={inputProps}
      inputRef={inputRef}
      multiline={multiline}
      name={name}
      onChange={onChange}
      placeholder={placeholder}
      readOnly={readOnly}
      required={required}
      rows={rows}
      rowsMax={rowsMax}
      startAdornment={startAdornment}
      type={type}
      value={value}
    />
  </StylesProvider>
)

RubiwinInputField.propTypes = {
  /** add a class to field */
  className: PropTypes.string,
  /** add an id to input */
  defaultValue: PropTypes.any,
  /** disable input */
  disabled: PropTypes.bool,
  /** error state */
  error: PropTypes.bool,
  /** add end adornment */
  endAdornment: PropTypes.node,
  /** fullWidth */
  fullWidth: PropTypes.bool,
  /** Attributes applied to the input element */
  inputProps: PropTypes.object,
  /** pass a ref to input element */
  inputRef: PropTypes.string,
  /** If true, a textarea element will be rendered. */
  multiline: PropTypes.bool,
  /** Name attribute of the input element. */
  name: PropTypes.string,
  /** onChange callback */
  onChange: PropTypes.func,
  /** placeholder */
  placeholder: PropTypes.string,
  /** readOnly */
  readOnly: PropTypes.bool,
  /** required */
  required: PropTypes.bool,
  /** Number of rows to display when multiline option is set to true. */
  rows: PropTypes.number,
  /** Maximum number of rows to display when multiline option is set to true. */
  rowsMax: PropTypes.number,
  /** add end adornment */
  startAdornment: PropTypes.node,
  /** Type of the input element. It should be a valid HTML5 input type. */
  type: PropTypes.string,
  /** value */
  value: PropTypes.any
}

export default RubiwinInputField
