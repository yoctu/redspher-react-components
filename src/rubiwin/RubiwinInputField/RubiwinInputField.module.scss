@import '../rubiwin-constant';

.root {
  background-color: white;
  border-radius: $border-radius-sm;
  border: 1px solid $dynamiteGrey;
  padding: 2px 6px;
  transition: $transition-all;
  &:hover {
    border: 1px solid darken($dynamiteGrey, 10%);
  }
}

.focused {
  border: 1px solid $primaryColor;
  border-radius: $border-radius-sm;
  transition: $transition-all;
  &:hover {
    border: 1px solid $primaryColor;
    border-radius: $border-radius-sm;
  }
}

.disabled {
  background-color: $disableGrey;
  cursor: not-allowed;
}

.error {
  border: 1px solid $danger;
  &:hover {
    border: 1px solid darken($danger, 5%);
  }
}
