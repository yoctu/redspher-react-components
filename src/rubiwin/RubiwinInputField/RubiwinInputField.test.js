import React from 'react'
import renderer from 'react-test-renderer'
import RubiwinInputField from './RubiwinInputField'
import MailIcon from '../../icons/MailIcon/MailIcon'

const testProps = {
  className: 'test',
  id: 'id',
  name: 'name',
  onChange: () => {},
  placeholder: 'placeholder',
  inputProps: { maxLength: 20 },
  type: 'text',
  value: 'rubiwin'
}

/** Snapshots */
it('renders correctly props object', () => {
  const tree = renderer.create(<RubiwinInputField {...testProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('is disabled', () => {
  const tree = renderer.create(<RubiwinInputField disabled />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('has error state', () => {
  const tree = renderer.create(<RubiwinInputField error />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('is full-width', () => {
  const tree = renderer.create(<RubiwinInputField fullWidth />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('has an input element added', () => {
  const tree = renderer
    .create(<RubiwinInputField inputProps={{ maxLength: 20 }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('is multiline, has 4 rows and 5 max rows', () => {
  const tree = renderer
    .create(<RubiwinInputField rows={5} rowsMax={6} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('is readOnly', () => {
  const tree = renderer.create(<RubiwinInputField readOnly />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('is required', () => {
  const tree = renderer.create(<RubiwinInputField required />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('has a start and end adornment', () => {
  const tree = renderer
    .create(
      <RubiwinInputField
        startAdornment={<MailIcon />}
        endAdornment={<MailIcon />}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
