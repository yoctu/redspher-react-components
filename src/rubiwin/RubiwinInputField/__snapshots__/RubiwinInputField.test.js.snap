// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a start and end adornment 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root MuiInputBase-adornedStart MuiInputBase-adornedEnd"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 15"
  >
    <path
      d="M18.4923 0H3.36049C0.88333 0 0 1.73305 0 3.36049V10.8592C0 12.4818 0.88333 14.2197 3.36049 14.2197H18.4923C20.1149 14.2197 21.8528 13.3364 21.8528 10.8592V3.36049C21.848 0.868927 20.1149 0 18.4923 0ZM3.36049 0.715305L18.4923 0.724906C19.0809 0.719866 19.6582 0.88675 20.1534 1.20498L10.9264 7.18185L8.74208 5.74164C8.7035 5.70648 8.65759 5.68025 8.60766 5.66483L1.68985 1.19537C2.18036 0.859326 2.76641 0.690881 3.36049 0.715305ZM0.724906 10.8592H0.729707V3.36049C0.731327 2.78183 0.897852 2.21564 1.20978 1.72825L7.93076 6.08729L1.46421 12.8083C0.979403 12.2759 0.715065 11.5792 0.724906 10.8592ZM18.4491 13.49L3.36529 13.4948C2.90899 13.5063 2.45604 13.4144 2.0403 13.2259L8.52605 6.48575L10.7296 7.92596H10.7536C10.8013 7.95002 10.8538 7.96311 10.9072 7.96437C10.9606 7.96311 11.0131 7.95002 11.0608 7.92596H11.0848L13.3027 6.48575L19.7837 13.2067C19.3642 13.3956 18.9091 13.4923 18.4491 13.49ZM20.427 12.7795L13.9604 6.05849L20.6814 1.70425C20.9976 2.1899 21.1536 2.76238 21.1279 3.34129V10.8592C21.1706 11.5688 20.9168 12.2642 20.427 12.7795Z"
    />
  </svg>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 15"
  >
    <path
      d="M18.4923 0H3.36049C0.88333 0 0 1.73305 0 3.36049V10.8592C0 12.4818 0.88333 14.2197 3.36049 14.2197H18.4923C20.1149 14.2197 21.8528 13.3364 21.8528 10.8592V3.36049C21.848 0.868927 20.1149 0 18.4923 0ZM3.36049 0.715305L18.4923 0.724906C19.0809 0.719866 19.6582 0.88675 20.1534 1.20498L10.9264 7.18185L8.74208 5.74164C8.7035 5.70648 8.65759 5.68025 8.60766 5.66483L1.68985 1.19537C2.18036 0.859326 2.76641 0.690881 3.36049 0.715305ZM0.724906 10.8592H0.729707V3.36049C0.731327 2.78183 0.897852 2.21564 1.20978 1.72825L7.93076 6.08729L1.46421 12.8083C0.979403 12.2759 0.715065 11.5792 0.724906 10.8592ZM18.4491 13.49L3.36529 13.4948C2.90899 13.5063 2.45604 13.4144 2.0403 13.2259L8.52605 6.48575L10.7296 7.92596H10.7536C10.8013 7.95002 10.8538 7.96311 10.9072 7.96437C10.9606 7.96311 11.0131 7.95002 11.0608 7.92596H11.0848L13.3027 6.48575L19.7837 13.2067C19.3642 13.3956 18.9091 13.4923 18.4491 13.49ZM20.427 12.7795L13.9604 6.05849L20.6814 1.70425C20.9976 2.1899 21.1536 2.76238 21.1279 3.34129V10.8592C21.1706 11.5688 20.9168 12.2642 20.427 12.7795Z"
    />
  </svg>
</div>
`;

exports[`has an input element added 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    maxLength={20}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`has error state 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root Mui-error Mui-error error"
  onClick={[Function]}
>
  <input
    aria-invalid={true}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`is disabled 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root Mui-disabled Mui-disabled disabled"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled disabled"
    disabled={true}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`is full-width 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root MuiInputBase-fullWidth MuiInput-fullWidth"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`is multiline, has 4 rows and 5 max rows 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    rows={5}
    type="text"
  />
</div>
`;

exports[`is readOnly 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    readOnly={true}
    type="text"
  />
</div>
`;

exports[`is required 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
  required={true}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
</div>
`;

exports[`renders correctly props object 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root test"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    id="id"
    maxLength={20}
    name="name"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="placeholder"
    type="text"
    value="rubiwin"
  />
</div>
`;
